java-checks/src/test/files/checks/StaticFieldInitializationCheck.java

```diff
@@ -148,4 +148,18 @@ public static void resetConfiguration2() {
     }
 
   }
+
+  class E {
```

Comment:
Try to keep the code you introduce as close as possible from something which compiles. It helps and is usually closer from real use cases. In particular, I modified your code as follow:

:warning: WARNING :warning: : I didn't checked if this code is still getting a 100% in term of code coverage.

```
static class E {
    static Config CONFIG;

    synchronized void foo() {
      CONFIG = null; // Compliant
    }

    synchronized Consumer<String> bar() {
      return s -> {
        CONFIG = null; // Noncompliant
      };
    }
  }
```